explorer/explorer.go

```diff
@@ -135,9 +140,50 @@ func (exp *explorerUI) rootWebsocket(w http.ResponseWriter, r *http.Request) {
 				exp.wsHub.HubRelay <- sigPingAndUserCount
 			}
 		}()
-
+		go func() {
+			for {
+				msg := &WebSocketMessage{}
+				websocket.JSON.Receive(ws, &msg)
+				var webData WebSocketMessage
+				switch msg.EventId {
+				case "decodetx":
+					webData.EventId = "decodedtx"
+					log.Debug("Recieved decodetx signal for hex: ", msg.Messsage)
+					tx := exp.blockData.DecodeRawTransaction(msg.Messsage)
+					if tx != nil {
+						buff := new(bytes.Buffer)
+						enc := json.NewEncoder(buff)
+						enc.SetIndent("", "    ")
+						enc.Encode(tx)
```

Comment:
This is basically the internals of `MarshalIndent`.  Might as well do this, and check for encode error:

```go
				message, err := json.MarshalIndent(webData.Messsage, "", "    ")
				if err != nil {
					log.Warn("Invalid JSON message: ", err)
					continue
				}
				webData.Messsage = string(message)```